docs/ai-agent/ai-agent-spans.md

```diff
@@ -0,0 +1,185 @@
+| ------------------------------ | ------ | ------------------------------------------ | -------------------------------- | ----------------- | --- |
+| `ai_agent.agent.name`          | string | Name of the agent.                         | `Researcher Bot`                 | Required          | ![Experimental](https://img.shields.io/badge/-experimental-blue)    |
+| `ai_agent.agent.role`          | string | Role assigned to the agent.                | `Data Collector`                 | Required          | ![Experimental](https://img.shields.io/badge/-experimental-blue)    |
+| `ai_agent.agent.backstory`     | string | Background story or context for the agent. | `Specializes in web data mining` | Required          | ![Experimental](https://img.shields.io/badge/-experimental-blue)    |
+| `ai_agent.agent.workflow_name` | string | Name of the workflow the agent is part of. | `Data Processing Pipeline`       | Required          | ![Experimental](https://img.shields.io/badge/-experimental-blue)    |
+| `ai_agent.agent.model`         | string | Underlying model powering the agent.       | `gpt-4`                          | Required          | ![Experimental](https://img.shields.io/badge/-experimental-blue)    |
+| `ai_agent.agent.tools`         | array  | List of tools available to the agent.      | `["Web Scraper", "Analyzer"]`    | Recommended       | ![Experimental](https://img.shields.io/badge/-experimental-blue)    |
+
+### Tools Attributes
+
+| Attribute                | Type   | Description                                  | Example           | Requirement Level | Stability    |
+| ------------------------ | ------ | -------------------------------------------- | ----------------- | ----------------- | --- |
+| `ai_agent.tool.name`     | string | Name of the tool utilized by the agent.      | `Web Scraper`     | Required          | ![Experimental](https://img.shields.io/badge/-experimental-blue)    |
+| `ai_agent.tool.function` | string | Specific function or capability of the tool. | `Data Extraction` | Recommended       | ![Experimental](https://img.shields.io/badge/-experimental-blue)    |
+| `ai_agent.tool.output`   | object | Output produced by the tool.                 | `{"data": [...]}` | Required          | ![Experimental](https://img.shields.io/badge/-experimental-blue)    |
+
+### Task Attributes
+
+| Attribute                   | Type    | Description                                                              | Example                            | Requirement Level | Stability    |
+| --------------------------- | ------- | ------------------------------------------------------------------------ | ---------------------------------- | ----------------- | --- |
+| `ai_agent.task.name`        | string  | Name of the task.                                                        | `Data Collection`                  | Required          | ![Experimental](https://img.shields.io/badge/-experimental-blue)    |
```

Comment:
do we also need a `ai_agent.task.input` in addition to these?

Reply:
I'm curious if we can consolidate it under `gen_ai.system|user|tool|assistant.message` events rather than attirbutes